packages/istanbul-lib-instrument/src/instrumenter.js

```diff
@@ -80,7 +80,13 @@ class Instrumenter {
         const opts = this.opts;
         const ast = babylon.parse(code, {
             allowReturnOutsideFunction: opts.autoWrap,
-            sourceType: opts.esModules ? "module" : "script"
+            sourceType: opts.esModules ? "module" : "script",
+            plugins: [
+              'asyncGenerators',
+              'dynamicImport',
+              'flow',
+              'jsx'
+            ]
```

Comment:
@bcoe This plugins are already part of babylon and flagged as experimental language proposal. These plugins are at least stage-0 proposal and activating them means enabling new parsing rules that are out of standard ES6 spec.

They tend to be newer feature, so yeah, the stability is ensured by the tests on the babylon repo but it won't be reliable as older feature.

Anyway this repository is already using this feature: https://github.com/istanbuljs/istanbuljs/blob/master/packages/istanbul-lib-instrument/src/read-coverage.js#L17

`plugins: ["*"]` is a soon to be deprecated way to call all babylon plugins. If you think that that part of code is stable then I suppose you can consider stable also this PR.